cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkContextConfig.java

```diff
@@ -16,191 +16,134 @@
+   */
+  public boolean isLocal() {
+    // In distributed mode, we run the drive in a twill container, hence the TWILL_INSTANCE_ID will be set.
+    return !System.getenv().containsKey("TWILL_INSTANCE_ID");
   }
 
-  public static void set(Configuration hadoopConf, BasicSparkContext context, CConfiguration conf, Transaction tx,
-                         Location programJarCopy) {
-    hConf = hadoopConf;
+  /**
+   * Sets configurations based on the given context.
+   */
+  public SparkContextConfig set(ExecutionSparkContext context) {
+    setSpecification(context.getSpecification());
+    setProgramId(context.getProgramId());
     setRunId(context.getRunId().getId());
     setLogicalStartTime(context.getLogicalStartTime());
-    //TODO: Change this once we start supporting Spark in Workflow
-    setProgramNameInWorkflow("Not Supported");
     setArguments(context.getRuntimeArguments());
-    setProgramJarName(programJarCopy.getName());
-    setProgramLocation(programJarCopy.toURI());
-    setConf(conf);
-    setTx(tx);
-  }
+    setTransaction(context.getTransaction());
 
-  private static void setArguments(Map<String, String> runtimeArgs) {
-    hConf.set(HCONF_ATTR_ARGS, new Gson().toJson(runtimeArgs));
+    return this;
   }
 
-  public Arguments getArguments() {
-    Map<String, String> arguments = new Gson().fromJson(hConf.get(HCONF_ATTR_ARGS),
-                                                        new TypeToken<Map<String, String>>() { }.getType());
-    return new BasicArguments(arguments);
+  /**
+   * Returns the {@link SparkSpecification} stored in the configuration.
+   */
+  public SparkSpecification getSpecification() {
+    return GSON.fromJson(hConf.get(HCONF_ATTR_PROGRAM_SPEC), SparkSpecification.class);
   }
 
-  public URI getProgramLocation() {
-    URI uri;
-    try {
-      uri = (new URI(hConf.get(HCONF_ATTR_PROGRAM_JAR_LOCATION)));
-    } catch (URISyntaxException use) {
-      LOG.error("Failed to create an URI from program location. The string violates RFC 2396", use);
-      throw Throwables.propagate(use);
-    }
-    return uri;
+  /**
+   * Returns the {@link Id.Program} stored in the configuration.
+   */
+  public Id.Program getProgramId() {
+    return GSON.fromJson(hConf.get(HCONF_ATTR_PROGRAM_ID), Id.Program.class);
   }
 
-  private static void setRunId(String runId) {
-    hConf.set(HCONF_ATTR_RUN_ID, runId);
+  /**
+   * Returns the {@link RunId} stored in the configuration.
+   */
+  public RunId getRunId() {
+    return RunIds.fromString(hConf.get(HCONF_ATTR_RUN_ID));
   }
 
-  public String getRunId() {
-    return hConf.get(HCONF_ATTR_RUN_ID);
-  }
-
-  private static void setLogicalStartTime(long startTime) {
-    hConf.setLong(HCONF_ATTR_LOGICAL_START_TIME, startTime);
+  /**
+   * Returns the runtime arguments stored in the configuration.
+   */
+  public Map<String, String> getArguments() {
+    return GSON.fromJson(hConf.get(HCONF_ATTR_ARGS), new TypeToken<Map<String, String>>() { }.getType());
```

Comment:
Just one usage in this class. It doesn't matter (in fact constant takes up perm-gen space).